x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/policy/AutoscalingPolicyMetadata.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.autoscaling.policy;
+
+import org.elasticsearch.cluster.AbstractDiffable;
+import org.elasticsearch.cluster.Diffable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class AutoscalingPolicyMetadata extends AbstractDiffable<AutoscalingPolicyMetadata>
+    implements
+        Diffable<AutoscalingPolicyMetadata>,
+        ToXContentObject {
+
+    static final ParseField POLICY_FIELD = new ParseField("policy");
+
+    @SuppressWarnings("unchecked")
+    public static final ConstructingObjectParser<AutoscalingPolicyMetadata, String> PARSER;
```

Comment:
I think this could be private?
```suggestion
    private static final ConstructingObjectParser<AutoscalingPolicyMetadata, String> PARSER;
```